src/components/studies/participants/BatchEditForm.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import { Dialog } from '@material-ui/core'
+import { EditParticipantForm } from './ParticipantForms'
+import { EditDialogTitle } from './ParticipantTableGrid'
+import ParticipantService from '../../../services/participants.service'
+
+type BatchEditFormProps = {
+  isEnrolledById: boolean
+  isBatchEditOpen: boolean
+  setIsBatchEditOpen: Function
+  selectedParticipants: string[]
+  token: string
+  studyId: string
+  toggleParticipantRefresh: Function
+  isAllSelected: boolean
+}
+
+const BatchEditForm: React.FunctionComponent<BatchEditFormProps> = ({
+  isEnrolledById,
+  isBatchEditOpen,
+  setIsBatchEditOpen,
+  selectedParticipants,
+  token,
+  studyId,
+  toggleParticipantRefresh,
+  isAllSelected,
+}) => {
+  const [isLoading, setIsLoading] = React.useState(false)
+
+  return (
+    <Dialog
+      open={isBatchEditOpen}
+      maxWidth="sm"
+      fullWidth
+      aria-labelledby="edit participant"
+    >
+      <EditDialogTitle
+        onCancel={() => {
+          setIsBatchEditOpen(false)
+        }}
+        shouldWithdraw={false}
+        batchEdit
+      />
+      <EditParticipantForm
+        isEnrolledById={isEnrolledById}
+        onCancel={() => {
+          setIsBatchEditOpen(false)
+        }}
```

Comment:
This is too long of a function to be inlined. Please have it as a separate function that takes an update type as an pargument